Render method declarations similarly to tests

toddtarsi opened the pull request:
### Description

This makes method body generation work more similarly to test generation.

Rather than using a one-off method to generate our method declarations, we take in the render and testLevel variables from emitTests, and use these to generate our method declarations.  

### Motivation and Context

The way it previously worked, command emitters that used any syntax other than raw strings would render as [Object object] within method declarations.
Standard emitters like this wouldn't work:

```
click: async function emitClick() {
  return promise.resolve({ commands: [{ statement: 'xyz.click()' level: 0 }] })
}
```

This meant that re-using tests as methods required some pretty heavy changes to the previously written export format.
Additionally, this meant that commands that affected indentation in methods, like control flow blocks, would generally fail to render correctly due to their usage of staringLevelAdjustment and endingLevelAdjustment.

### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist
- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium-ide/blob/master/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/selenium-ide?pullRequest=1223) <br/>All committers have signed the CLA.

toddtarsi commented:
Also, I'm not sure what's going on with the CLA stuff. I signed it so long ago that it's saying I haven't signed it or something? When I go to sign it again, it tells me I already signed.

toddtarsi committed:
Render method declarations similarly to tests

The way it previously worked, commands that used any syntax other than strings would render as [Object object].
This meant that commands that were designed to indent, like control flow blocks, had to basically not do all of that to work.

Also, this adds control flow commands to the reuse case test as well to ensure this works in future versions

toddtarsi commented:
> Also, I'm not sure what's going on with the CLA stuff. I signed it so long ago that it's saying I haven't signed it or something? When I go to sign it again, it tells me I already signed.

NM, I needed to set my email in my local git config. CLA stuff passes now, thanks!

toddtarsi commented:
Looking at this option further, it seems like I probably want something for method indentation.

Also, if this is preferred, I can point this PR at master instead since it deals more with side-utils than the ide.

toddtarsi committed:
[side-utils]: emitMethod - prevent double \n

toddtarsi committed:
[test] update snapshots on generated code

toddtarsi committed:
no-op comment change to restart the build tests

AutomatedTester commented on packages/code-export-python-pytest/__test__/src/__snapshots__/index.spec.js.snap:
```diff
@@ -305,7 +306,8 @@ class TestValidcredentials():
     self.driver.find_element(By.ID, \\"username\\").send_keys(self.vars[\\"username\\"])
     self.driver.find_element(By.ID, \\"password\\").send_keys(self.vars[\\"password\\"])
-    self.driver.find_element(By.CSS_SELECTOR, \\".fa\\").click()
+    if self.driver.execute_script(\\"return (true)\\"):
```
Could you explain why we're doing this now? I don't understand this change

toddtarsi replied:
@AutomatedTester - Yessir, the reason I added this logic is that my fixes target control flow logic statements more than anything else.

If you add control flow logic to a reused method currently, the if and end blocks will render as `[object Object]`, breaking code export in all languages. You can try it by using the updated side file in this branch in master, and export to any language. It will be 100% broken.

Not to get too long winded, but this is because method bodies were generated using a one-off method that didn't handle all cases like emitCommands does.

So I figured, in addition to writing these fixes, I would expand the test SIDE files so that control flow logic were included, and any future regressions would be caught. This was the simplest possible implementation of control flow logic:

```
if true
    click xyz
end
```

After adjusting the side file, the snapshots were failing because the new steps weren't there, so I had to update the snapshots as well.

Okay, that was a long explanation, but yeah.
